src/libraries/Common/src/System/SR.cs

```diff
@@ -10,14 +10,20 @@ namespace System
 {
     internal partial class SR
     {
+#if (!NETSTANDARD1_0 && !NETSTANDARD1_1 && !NET45) // AppContext is not supported on < NetStandard1.3 or < .NET Framework 4.5
+        private static readonly bool s_usingResourceKeys = AppContext.TryGetSwitch("System.Resources.UseResourceKeys", out bool usingResourceKeys) ? usingResourceKeys : false;
```

Comment:
I'm not sure this is a good change, it adds complex static cctor to resources init path. Wouldn't be better to use LazyInitializer.EnsureInitialized

Reply:
LazyInitializer.EnsureInitialized is more expensive that this, on CoreCLR at least. It adds lambda, delegate to wrap the lambda in and a field to cache the lambda.